### PR TITLE
Added Kokkos profiling to Spot controller

### PR DESCRIPTION
**Pull request by DavidPoliakoff**
I'm not 100% sure this is how to make that query string, feel free to correct me if stuff is broken. I also don't know whether appending the service might want to be done at a higher level (the base controller), let me know on that as well

### CODE REVIEW

File: src/mpi-rt/controllers/SpotController.cpp [hunk 1]
```diff
@@ -232,6 +232,9 @@ class SpotController : public cali::ChannelController
             if (opts.get("output", "").to_string() != "adiak")
                 config()["CALI_SERVICES_ENABLE"].append(",adiak_import");
 #endif
+            if (opts.get("profile.kokkos","").to_bool()) {
+                config()["CALI_SERVICES_ENABLE"].append(",kokkostime");
+            }
             m_opts.update_channel_config(config());
```
Comment (daboehme) [inline src/mpi-rt/controllers/SpotController.cpp]: Won't need that, the options take care of the services automatically

File: src/caliper/controllers/controllers.cpp [hunk 2]
```diff
@@ -45,6 +45,14 @@ const char* builtin_option_specs =
     " \"query args\"  : { \"level\": \"runtime\", \"group by\": [ \"cupti.runtimeAPI\" ] }"
     "},"
     "{"
+    " \"name\"        : \"profile.kokkos\","
+    " \"type\"        : \"bool\","
+    " \"description\" : \"Profile Kokkos functions\","
+    " \"category\"    : \"region\","
+    " \"services\"    : [ \"kokkostime\"  ],"
+    " \"query args\"  : { \"level\": \"runtime\", \"group by\": [ \"function\" ] }"
```
Comment (daboehme) [inline src/caliper/controllers/controllers.cpp]: Can skip the query args here. This group by is only needed if you have some custom attribute; if you're using the regular "function" it's fine. This thing is admittedly a bit weird.

Comment (DavidPoliakoff) [reply src/caliper/controllers/controllers.cpp]: If I leave query args as an empty dict, it has a problem. Should I just delete the query args field?

Comment (daboehme) [reply src/caliper/controllers/controllers.cpp]: You can just delete it.

**Comment by DavidPoliakoff** [comment]
Cool, your changes worked in my test. This is just a fantastic system, the ConfigManager.

File: src/mpi-rt/controllers/SpotController.cpp (hunk 1 above)
Comment (DavidPoliakoff) [reply src/mpi-rt/controllers/SpotController.cpp]: Addressed

File: src/caliper/controllers/controllers.cpp (hunk 2 above)
Comment (DavidPoliakoff) [reply src/caliper/controllers/controllers.cpp]: Addressed

**Pull request merged**